### PR TITLE
[SW-197] Make spot ros2 properly load arm urdf and arm frames for spot with arm

### PR DESCRIPTION
**Pull request by kaiyu-zheng**
This PR fixes URDF loading and TF frame publishing issues for Spot with arm. Specifically:

* Adds missing mappings for arm frames in ros_helpers
* Corrects [spot_description/urdf/spot_with_arm.urdf.xacro](https://github.com/bdaiinstitute/spot_ros2/compare/SW-197-Make-spot_ros2-properly-load-arm-URDF-and-arm-frames-for-Spot-with-arm?expand=1#diff-59663014c6c3936ee577f6bef02c6069b4512883bcc5e00231f064e53c20768e) to be using `spot.urdf.xacro` instead of `spot_simple.urdf.xacro`.
* Adds `has_arm` argument to driver launch files (both with and without namespace)

Note that the last bullet point is a temporary solution for completeness - it enables configuring the driver to switch between loading URDF with and without arm in a simple way, which is currently not possible.

This PR does not address https://github.com/bdaiinstitute/spot_ros2/issues/21, but it does address https://github.com/bdaiinstitute/spot_ros2/issues/43#issuecomment-1538781073  in terms of adding support to Spot with arm.   This PR is worth merging given the other critical fixes. Implementing https://github.com/bdaiinstitute/spot_ros2/issues/21 is the right thing to do but should be a separate PR.
 
Launching the driver with `has_arm:=True` should enable visualizing the full URDF and streaming arm TF frames.

### CODE REVIEW

File: spot_driver/launch/spot_driver_with_namespace.launch.py [hunk 1]
```diff
@@ -3,23 +3,48 @@
...
+    )
+    ld.add_action(robot_state_publisher)
+
+
 def generate_launch_description() -> launch.LaunchDescription:
     spot_name = LaunchConfiguration("spot_name")
     spot_name_arg = DeclareLaunchArgument("spot_name", description="Name of spot")
 
+    has_arm = LaunchConfiguration("has_arm")
+    has_arm_arg = DeclareLaunchArgument("has_arm", description="Name of spot", default_value="False")
```
Comment (amessing-bdai) [inline spot_driver/launch/spot_driver_with_namespace.launch.py:42]: Fix Description

**Comment by kaiyu-zheng** [comment]
> Can you edit the readme to mention this argument?

Done.


> Also, out of curiosity, what happens if you set the argument on a robot without an arm?

Good question. Just tried. This is what I get in RViZ -- a reasonable failure state

image

**Comment by kaiyu-zheng** [comment]
@farbod-farshidian added a README under spot_descriptions/urdf and filled in what I know.

https://github.com/bdaiinstitute/spot_ros2/blob/906738ed792b534b9f6272789e27f30ab181c38c/spot_description/urdf/README.md

**Comment by lfermoselle-bdai** [comment]
I have HIL tested this PR using a Spot with an arm.

With `has_arm:=True` argument, I could successfully visualize in Rviz the URDF of Spot with the arm (see screenshot below)
Screenshot from 2023-06-14 17-29-50

With `has_arm:=False`, the URDF visualized in Rviz had no arm, as intended (see screenshot below):
Screenshot from 2023-06-14 17-30-08

**Review by amessing-bdai — APPROVED** [review]

**Pull request merged**